go/vt/mysqlctl/mysqld.go

```diff
@@ -644,28 +662,16 @@ func getMycnfTemplates(root string) []string {
 		cnfTemplatePaths = append(cnfTemplatePaths, parts...)
 	}
 
-	switch mysqlFlavor := os.Getenv("MYSQL_FLAVOR"); mysqlFlavor {
-	case "MariaDB":
-		path := path.Join(root, "config/mycnf/master_mariadb.cnf")
-		if !contains(cnfTemplatePaths, path) {
-			cnfTemplatePaths = append(cnfTemplatePaths, path)
-		}
-	case "MariaDB103":
-		path := path.Join(root, "config/mycnf/master_mariadb103.cnf")
-		if !contains(cnfTemplatePaths, path) {
-			cnfTemplatePaths = append(cnfTemplatePaths, path)
-		}
-	case "MySQL80":
-		path := path.Join(root, "config/mycnf/master_mysql80.cnf")
-		if !contains(cnfTemplatePaths, path) {
-			cnfTemplatePaths = append(cnfTemplatePaths, path)
-		}
-	default:
-		path := path.Join(root, "config/mycnf/master_mysql56.cnf")
-		// By default we assume Mysql56 compatable
-		if !contains(cnfTemplatePaths, path) {
-			cnfTemplatePaths = append(cnfTemplatePaths, path)
-		}
+	// master_{flavor}.cnf
+	p := path.Join(root, fmt.Sprintf("config/mycnf/master_%s.cnf", mysqld.flavor))
```

Comment:
We need to be careful adding new files here. We can't assume everyone running this new code will have copied the new files in `config/mycnf` into their production config dir.

For files that already existed before this PR, like `master_mariadb.cnf`, we should be fine to require them. But for new files, like `master_mysql.cnf` and `master_mysql57.cnf`, we should make them optional. We could do that by checking if those files exist before adding them to the list. In general, we still want `makeMycnf` to bail out if anything in the list is not found, for safety reasons.

Reply:
I have changed the behavior, but want to run it by you. If a `EXTRA_MY_CNF` is specified, it must exist.  But `master_mysql57.cnf` etc are treated like a search path (vitess will look in these locations, but is okay with the files not existing). This config file handling is mysql-like, and retrofits well.. but not for everyone.